File: reinvent-2019/learn-languages-ai-ml/Buildersfair-API/Models/StageLog.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BuildersFair_API.Models
{
    [Table("tb_stage_log")]
    public class StageLog
    {
        public int game_id { get; set; }
        public int stage_id { get; set; }
        public int objects_score { get; set; }
        public int time_score { get; set; }
        public int clear_score { get; set; }
        public int stage_score { get; set; }
        public int total_score { get; set; }
        public string completed_yn { get; set; } // Y/N
        public DateTime start_date { get; set; } 
        public DateTime? end_date { get; set; } 
    }
}

File: reinvent-2019/learn-languages-ai-ml/Buildersfair-Web/src/app/_models/stagelog.ts
export interface StageLog {
    game_id: number;
    stage_id: number;
    objects_score: number;
    time_score: number;
    clear_score: number;
    stage_score: number;
    total_score: number;
    completed_yn: string;
    start_date: Date;
    end_date: Date;
}
